mlir_graphblas/src/include/GraphBLAS/GraphBLASOps.td

```diff
@@ -777,6 +777,40 @@ def GraphBLAS_CommentOp : GraphBLAS_Op<"comment", []> {
     }];
 }
 
+def GraphBLAS_PrintOp : GraphBLAS_Op<"print", []> {
+    let summary = "graphblas print operation";
+    let description = [{
+        `graphblas.print` is used to pretty print values
+        to stdout.
+        This is intended to be used for debugging only.
+        The strings attribute is a list of strings.
+        This op is variadic and takes an arbitrary number of inputs.
+        The printing alternates between printing the strings
+        and the input values. 
+
+        ```mlir 
+        %c9_9_f32 = constant 9.9 : f32
+        %c1_i32 = constant 1 : i32
+        // prints "start 9.9 middle 1 end".
+        graphblas.print %c9_9_f32, %c1_i32 { strings = ["start ", " middle ", " end"] } : f32, i32
+        // prints "start 9.9 middle end zyx".
+        graphblas.print %c9_9_f32 { strings = ["start ", " middle ", " end", " z", "y", "x"] } : f32
+        // prints "start 9.91111".
+        graphblas.print %c9_9_f32, %c1_i32, %c1_i32, %c1_i32, %c1_i32 { strings = ["start "] } : f32, i32, i32, i32, i32
```

Comment:
I find this non-intuitive. If no string is provided for a value, we should at least print a space to separate the values. Is there a valid use case where doing a "numeric concatenation" is useful?

I would argue that the same should be applied if there are extra strings beyond the values. They should be separated by a space. Pure concatenation is unexpected.